tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py

```diff
@@ -170,11 +179,24 @@ def _forward_backward_test_impl(
                 total_layers = pipeline_model_parallel_world_size
                 if virtual_pipeline_model_parallel_size is not None:
                     total_layers *= virtual_pipeline_model_parallel_size
-                target_loss = get_target_loss(global_batch_shape, hidden_size, total_layers)
+                target_loss, target_model = get_target_loss_and_model(global_batch_shape, hidden_size, total_layers)
 
                 for loss_item in loss:
                     x = loss_item['avg']
-                    torch.testing.assert_close(x / microbatch_size, target_loss*torch.ones_like(x))
+                    torch.testing.assert_close(x[0].cpu() / microbatch_size, target_loss)
+
+                if not forward_only:
+                    for vm_id, model_module in enumerate(model):
+                        params = list(model_module.parameters())
+                        rank = params[0].get_device()
+                        offset = pipeline_model_parallel_world_size
+                        param_id = (rank * offset) // self.world_size + vm_id * offset
```

Comment:
Do we want `self.world_size` or would just `pipeline_model_parallel_world_size` be ok?

Reply:
Agree, a better expression would be `param_id = rank // data_parallel_size + vm_id * offset`. I will commit the change when other conversations are resolved.